GroupManager/Utils/MsalAppBuilder.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using Microsoft.Identity.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Microsoft.Identity.Client.AppConfig;
+
+namespace GroupManager.Utils
+{
+    public static class MsalAppBuilder
+    {
+        public static IConfidentialClientApplication BuildConfidentialClientApplication()
+        {
+            IConfidentialClientApplication clientapp = ConfidentialClientApplicationBuilder.Create(Globals.ClientId)
+                  .WithClientSecret(Globals.ClientSecret)
+                  .WithRedirectUri(Globals.RedirectUri)
+                  .WithAuthority(new Uri(Globals.Authority))
+                  .Build();
+
+            MSALPerUserMemoryTokenCache userTokenCache = new MSALPerUserMemoryTokenCache(clientapp.UserTokenCache);
+            MSALAppMemoryTokenCache appTokenCache = new MSALAppMemoryTokenCache(clientapp.AppTokenCache, Globals.ClientId);
```

Comment:
Do we use it in this application? Is there a client creds usage?

Reply:
Not in this application, but I added it as an example. Shall i remove this?

---
In reply to: [271643544](https://github.com/Azure-Samples/active-directory-dotnet-admin-restricted-scopes-v2/pull/16#discussion_r271643544) [](ancestors = 271643544)

Reply:
yes please